### PR TITLE
feat: apply alphabetical ordering for languages

### PR DESCRIPTION
**Pull request by pnal**
https://oat-sa.atlassian.net/browse/AUT-1249
How to test:
- [x] go to back office Lists page, check language list values order
- [x] check that language list is not editable

Related PR: https://github.com/oat-sa/tao-core/pull/3348
- [x] TODO: update tao-core dependency version

### CODE REVIEW

File: controller/Lists.php [hunk 1]
```diff
@@ -531,14 +539,24 @@ private function getListData(bool $showRemoteLists = false): array
...
-                'elements' => $listElements->jsonSerialize(),
+                'elements' => $this->getSortedElementsDependingOnListClass($listClass, $listElements),
                 'totalCount' => $listElements->getTotalCount(),
             ];
         }
 
         return $lists;
     }
 
+    private function getSortedElementsDependingOnListClass(
```
Comment (bartlomiejmarszal) [inline controller/Lists.php:550]:
```suggestion
    private function sortElementsByListClass(
```

**Review by bartlomiejmarszal — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: controller/Lists.php [hunk 2]
```diff
@@ -531,14 +539,24 @@ private function getListData(bool $showRemoteLists = false): array
                 'uri' => tao_helpers_Uri::encode($listClass->getUri()),
                 'label' => $listClass->getLabel(),
                 'editable' => $listService->isEditable($listClass),
-                'elements' => $listElements->jsonSerialize(),
+                'elements' => $this->getSortedElementsDependingOnListClass($listClass, $listElements),
```
Comment (andreluizmachado) [inline controller/Lists.php:542]: please do not forget to update tao-core on composer.json file.

File: controller/Lists.php [hunk 3]
```diff
@@ -319,10 +323,13 @@ public function saveLists(ValueCollectionService $valueCollectionService): void
             )
         );
 
-        $listElements = array_filter($payload, function (string $key) {
-            return (bool)preg_match('/^list-element_/', $key);
-        },
-        ARRAY_FILTER_USE_KEY);
+        $listElements = array_filter(
+            $payload,
+            function (string $key) {
```
Comment (andreluizmachado) [inline controller/Lists.php:328]: `function (string $key): bool`

**Review by andreluizmachado — CHANGES_REQUESTED** [review]

File: controller/Lists.php (hunk 1 above)
Comment (bartlomiejmarszal) [reply controller/Lists.php:550]: Can we consider renaming this method?

File: controller/Lists.php (hunk 3 above)
Comment (pnal) [reply controller/Lists.php:328]: No problem, but you know, I did not touch it most of the changes related to cs fixer formatting.

File: controller/Lists.php (hunk 2 above)
Comment (pnal) [reply controller/Lists.php:542]: I have this item in checklist in PR description in order to not forget to do it when I'd be ready to merge.

File: controller/Lists.php (hunk 1 above)
Comment (pnal) [reply controller/Lists.php:550]: Why?

File: controller/Lists.php [hunk 4]
```diff
@@ -319,10 +323,13 @@ public function saveLists(ValueCollectionService $valueCollectionService): void
             )
         );
 
-        $listElements = array_filter($payload, function (string $key) {
-            return (bool)preg_match('/^list-element_/', $key);
-        },
-        ARRAY_FILTER_USE_KEY);
+        $listElements = array_filter(
```
Comment (andreluizmachado) [inline controller/Lists.php:326]: is it possible to increase the coverage of automated tests of this class?

Comment (pnal) [reply controller/Lists.php:326]: I do not think it is feasible to cover all the controller by test in scope of this ticket. Also I have seen a task for redesign the lists index page https://oat-sa.atlassian.net/browse/ADF-790 in an epic https://oat-sa.atlassian.net/browse/ADF-792 I suppose those are better candidates to include improving the test coverage for that area into their scope.

File: controller/Lists.php (hunk 2 above)
Comment (pnal) [reply controller/Lists.php:542]: https://github.com/oat-sa/extension-tao-backoffice/pull/127/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R26

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by bartlomiejmarszal — APPROVED** [review]
LGTM

**Review by bartlomiejmarszal — APPROVED** [review]
LGTM

**Pull request merged**